### PR TITLE
fix(fossid-webapp): Handle breaking change in the delete scan function

### PR DESCRIPTION
**Pull request by nnobelis**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: clients/fossid-webapp/src/test/kotlin/FossId2023dot1Test.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.ossreviewtoolkit.clients.fossid.deleteScan
+
+private const val PROJECT_CODE = "semver4j"
+private const val SCAN_CODE_2021_2 = "${PROJECT_CODE}_20201203_090342"
+
+/**
+ * This client test tests the calls that have been changed for FossID 2023.1.
+ * This version of FossID changed the return type of the deleteScan function: it was before an integer, and it is now an
+ * map of strings to strings. Creating a special [FossIdServiceWithVersion] implementation for this call is an overkill
+ * as ORT does not even use the return value. Therefore, this change is handled by the Jackson deserializer.
```
Comment (sschuberth) [inline clients/fossid-webapp/src/test/kotlin/FossId2023dot1Test.kt:41]: At least this part of the docs to me sound like something that should go to the deserializer, not to this test.

**Comment by sschuberth** [comment]
> Test failures are unrelated as they are from getVulnerabilitiesXXX functions.

This has already been fixed, please rebase.

File: clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt [hunk 2]
```diff
@@ -153,6 +157,19 @@ interface FossIdRestService {
...
+                            String::class.java,
+                            String::class.java
+                        )
+                        val map = JSON_MAPPER.readValue<Map<Any, Any>>(p, mapType)
+                        if (map.size != 1) {
+                            error("A map representing a polymorphic integer should have one value!")
+                        }
+                        PolymorphicInt(map.values.first().toString().toInt())
+                    }
+
```
Comment (sschuberth) [inline clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt:172]: Nit: Either this empty line should go away (like it was before), or empty lines between all cases should be added.

File: clients/fossid-webapp/src/test/kotlin/FossId2023dot1Test.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import com.github.tomakehurst.wiremock.WireMockServer
```
Comment (sschuberth) [inline clients/fossid-webapp/src/test/kotlin/FossId2023dot1Test.kt:20]: Missing `package` directive.

File: clients/fossid-webapp/src/test/kotlin/FossId2021dot2Test.kt [hunk 4]
```diff
@@ -25,6 +25,10 @@ import io.kotest.matchers.nulls.shouldNotBeNull
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.types.shouldBeInstanceOf
 
+import io.mockk.clearAllMocks
```
Comment (sschuberth) [inline clients/fossid-webapp/src/test/kotlin/FossId2021dot2Test.kt:28]: Commit message: "Mock*K*"

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: clients/fossid-webapp/src/test/kotlin/FossId2023dot1Test.kt (hunk 3 above)
Comment (nnobelis) [reply clients/fossid-webapp/src/test/kotlin/FossId2023dot1Test.kt:20]: Other tests in this project are also missing a `package`. I will add them in a separate commit.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**